File: ql_lich_giang_day_app/lib/api/auth_service.dart
import '../api/api_client.dart';

class AuthService {
  /// 🟢 Đăng nhập vào hệ thống
  static Future<Map<String, dynamic>> login(
    String tenDangNhap,
    String matKhau,
  ) async {
    final res = await ApiClient.post('login', {
      'TenDangNhap': tenDangNhap,
      'MatKhau': matKhau,
    });

    if (res['token'] != null) {
      ApiClient.setToken(res['token']);
    }

    // 🟦 Xử lý lấy role (vai trò)
    String? role;
    if (res['user'] != null) {
      role =
          res['user']['TenVaiTro'] ??
          (res['user']['DanhSachVaiTro']?.isNotEmpty == true
              ? res['user']['DanhSachVaiTro'][0]['TenVaiTro']
              : null);
    }

    // 🟩 Trả về dữ liệu gọn gàng cho màn hình Login
    return {'token': res['token'], 'user': res['user'], 'role': role};
  }

  /// 🟢 Lấy thông tin người dùng hiện tại (đã đăng nhập)
  static Future<Map<String, dynamic>> me() async {
    return await ApiClient.get('me');
  }

  /// 🟢 Đăng xuất tài khoản
  static Future<void> logout() async {
    await ApiClient.post('logout', {});
    ApiClient.clearToken();
  }
}

File: fe_tlu_web/src/api/services/authService.js
// src/api/services/pdt/authService.js
import { apiClient } from "../apiClient";

export const AuthService = {
    async login(username, password) {
        const res = await apiClient.post("/login", {
            TenDangNhap: username,
            MatKhau: password,
        });

        const data = res.data;

        // 🧹 Xóa token cũ (nếu có)
        localStorage.removeItem("token");
        localStorage.removeItem("roles");
        localStorage.removeItem("user");

        // 💾 Lưu token mới
        localStorage.setItem("token", data.token);
        localStorage.setItem("roles", JSON.stringify(data.roles));
        localStorage.setItem("user", JSON.stringify(data.user));

        return data;
    },
    async getProfile() {
        const res = await apiClient.get("/user");
        return res.data;
    },
};

// export const AuthService = {
//     async login(username, password) {
//         const res = await apiClient.post("/login", {
//             TenDangNhap: username,
//             MatKhau: password,
//         });

//         const data = res.data;

//         // 🧩 Lấy danh sách vai trò từ user
//         const roles = data.user?.DanhSachVaiTro?.map((v) => v.TenVaiTro) || [
//             data.user?.TenVaiTro,
//         ];

//         // 💾 Lưu vào localStorage
//         localStorage.setItem("token", data.token);
//         localStorage.setItem("user", JSON.stringify(data.user));
//         localStorage.setItem("roles", JSON.stringify(roles));

//         return { ...data, roles }; // ✅ trả roles ra ngoài
//     },

//     async getProfile() {
//         const res = await apiClient.get("/user");
//         return res.data;
//     },
// };
